desktop/cmp/grid/impl/colchooser/ColChooserModel.ts

```diff
@@ -111,19 +114,33 @@ export class ColChooserModel extends HoistModel {
     //------------------------
     syncChooserData() {
         const {gridModel, lrModel} = this,
-            columns = gridModel.getLeafColumns(),
-            hasGrouping = columns.some(it => it.chooserGroup);
+            hasGrouping = gridModel.getLeafColumns().some(it => it.chooserGroup),
+            columnState = sortBy(gridModel.columnState, it => {
+                const {pinned} = it;
+                if (pinned === 'left') {
+                    return 0;
+                }
+
+                if (pinned === 'right') {
+                    return 2;
+                }
+
+                return 1;
+            });
```

Comment:
This sort wasn't super clear to me - is the idea that our `columnState` is already in the correct order unless there are pinned cols, which aren't accounted for in that ordering?

Could comment that to clarify.  (Also can't help but make me wonder if this should be handled somehow at a deeper level - seems "funny" to have to deal with this concern here, but not clear to me we had any particular contract around sorting out of `columnState`)

Reply:
Yeah when a column is pinned it's order in column state remains where it was before becoming pinned, and the relative order of pinned columns on the same side is correct in column state but where exactly they are relative to other columns not pinned or not on the same side is somewhat undefined, this is just how ag-grid handles things, I assume to keep the column state as a simple list, and to get the behavior of columns going back to their original position when the user unpins them (which I guess makes sense)

Will add a comment to clarify for the next person to encounter that code